src/iop/basecurve.c

```diff
@@ -329,7 +329,7 @@ static void set_presets(dt_iop_module_so_t *self, const basecurve_preset_t *pres
     dt_gui_presets_update_ldr(_(presets[k].name), self->op, self->version(), FOR_RAW);
     // make it auto-apply for matching images:
     dt_gui_presets_update_autoapply(_(presets[k].name), self->op, self->version(),
-                                    force_autoapply ? *force_autoapply : presets[k].autoapply);
+                                    force_autoapply && presets[k].autoapply);
```

Comment:
you've changed the semantic. to me the pointer was correct and the line above we to be read:

- if force_autoapply is not null
   => for apply if force_autoapply is 1 (*force_autoapply)
   => otherwise apply if preset.autoapply is set.

Of course we can do the same with a standard int or even better a gboolean and this is probably better than having this pointer.

Reply:
Well the output was always TRUE because the preset was applied no matter what, so the global preference was actually useless. That's what I know. What happens in the piping is still a bit unclear to me, but the pointer seemed unnecessary so this was the faster way to fix the bug I found.

Reply:
Sure the pointer is not needed. We can use a gboolean (better than int here) as I said. Maybe we can start with that (code clean-up) and try to understand the other issue after?